### PR TITLE
Documentation for Code Validation in Tutorials

### PR DESCRIPTION
**Pull request by thsparks**
Figured I'd throw something together quickly, just so we have this information down somewhere. If we add more validators/properties in the future, it may be worth pulling this out of "Basics" and into its own file, but I didn't think that was necessary right out of the gate. I don't feel strongly, though, so if people would prefer to put it in its own section, let me know.

### CODE REVIEW

File: docs/writing-docs/tutorials/basics.md [hunk 1]
```diff
@@ -267,6 +267,33 @@ Gray blocks will be generated when not all datatypes are known (e.g. `mySprite`
...
 
+## Code Validation (`validation.local` and `validation.global` sections)
+
+If you want to enable code validation in your tutorial, you can do so by adding a validation section. As with [`blockconfig` above](#reconfiguring-blocks-in-the-toolbox-blockconfiglocal-and-blockconfigglobal-sections), there are two types of sections: `validation.global` and `validation.local`. Global can be anywhere in the markdown and applies to all steps in the tutorial. Local must appear within a specific step, and applies only to that step. Local takes precedence over global.
+
+Within a validation section, you may specify which validators you want to enable and properties for those validators using the same syntax we use for writing Skillmaps (see [skillmap structure](..\skillmaps.md#skillmap-structure)).
+
+### Validators
+
+Currently, only one validator exists: the `BlocksExistValidator`. This validator looks at highlighted blocks in the answer key and confirms that, for each highlighted block, the user's code contains at least one block of the same type. It does *not* validate the parameters passed into the block.
```
Comment (thsparks) [inline docs/writing-docs/tutorials/basics.md:278]: I couldn't find any docs on highlights in the answer key. Do we have those? If so, I'd like to link to them here.

Comment (kiki-lee) [reply docs/writing-docs/tutorials/basics.md:278]: I don't know if we do.  We should probably add //@highlight (and if it does exist, update to explain that you can now highlight multiple blocks). 

Related, do we have documentation on @hide and @collapse?

File: docs/writing-docs/tutorials/basics.md [hunk 2]
```diff
@@ -267,6 +267,33 @@ Gray blocks will be generated when not all datatypes are known (e.g. `mySprite`
...
+
+The only property currently available on the `BlocksExistValidator` is `Enabled`, which determines whether or not the validator runs.
+
+### Examples
+**Enable the `BlocksExistValidator` globally**
+````
+```validation.global
+# BlocksExistValidator
```
Comment (kimprice) [inline docs/writing-docs/tutorials/basics.md:286]: Do you need to include `Enabled: true` or is that set by default?

Comment (thsparks) [reply docs/writing-docs/tutorials/basics.md:286]: Set by default. I updated the docs to make that clearer, thanks!

File: docs/writing-docs/tutorials/basics.md (hunk 1 above)
Comment (jwunderl) [reply docs/writing-docs/tutorials/basics.md:278]: looks like highlight is mentioned here https://makecode.com/writing-docs/snippets#highlight, I should add hide /collapse there as well

**Review by jwunderl — APPROVED** [review]

**Review by kimprice — APPROVED** [review]

File: docs/writing-docs/tutorials/basics.md (hunk 1 above)
Comment (thsparks) [reply docs/writing-docs/tutorials/basics.md]: Nice, thanks! I've added a link to that section.

**Pull request merged**